analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SupplyParameterFromQueryNeedRoutableComponent_Partial.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Microsoft.AspNetCore.Components;
+using System;
+
+namespace EmptyProject
+{
+    public partial class SupplyParameterFromQueryNeedRoutableComponent_Partial : ComponentBase
+    {
+        [Parameter]
+        [SupplyParameterFromQuery]
+        public TimeSpan TimeSpan { get; set; } // Compliant
```

Comment:
I would explain after `// Compliant` that the reason for this being compliant is that the razor part of the component defines the route.